src/resultsutil.h

```diff
@@ -51,13 +54,23 @@ void setupHeaderView(QTreeView* view);
 
 void connectFilter(QLineEdit* filter, QSortFilterProxyModel* proxy);
 
-void setupTreeView(QTreeView* view, QLineEdit* filter, QAbstractItemModel* model, int initialSortColumn, int sortRole,
-                   int filterRole);
+template<typename Results, typename ModelImpl>
+void setupTreeView(QTreeView* view, QLineEdit* filter, QAbstractItemModel* model, int initialSortColumn, int sortRole)
+{
+    auto* proxy = new CostProxy<Results, ModelImpl>(view);
+    proxy->setSortRole(sortRole);
+    proxy->setSourceModel(model);
+    connectFilter(filter, proxy);
+
+    view->sortByColumn(initialSortColumn, Qt::DescendingOrder);
+    view->setModel(proxy);
+    setupHeaderView(view);
+}
 
-template<typename Model>
+template<typename Results, typename ModelImpl, typename Model>
 void setupTreeView(QTreeView* view, QLineEdit* filter, Model* model)
 {
-    setupTreeView(view, filter, model, Model::InitialSortColumn, Model::SortRole, Model::FilterRole);
+    setupTreeView<Results, ModelImpl>(view, filter, model, Model::InitialSortColumn, Model::SortRole);
```

Comment:
create the templated proxy here, then delegate to the implementation above (which shouldn't be inline)

all of this reduces template bloat